knime_extension/src/nodes/spatialnetwork.py

```diff
@@ -868,7 +878,292 @@ def ckdnearest(gdA, gdB):
             df = pd.DataFrame(data)
             dff = pd.concat([df, dff], ignore_index=True)
 
-        dff["DestinationID"] = dff.DestinationID - gdfmax
-        dff.sort_values(by=["OriginID", "DestinationID"], inplace=True)
+        dff["destinationid"] = dff.destinationid - gdfmax
+        dff.sort_values(by=["originid", "destinationid"], inplace=True)
         dff = dff.reset_index(drop=True)
         return knext.Table.from_pandas(dff)
+
+
+############################################
+# Isochrone map
+############################################
+@knext.node(
+    name="Isochrone Map",
+    node_type=knext.NodeType.MANIPULATOR,
+    # node_type=knext.NodeType.MANIPULATOR,
+    category=__category,
+    icon_path=__NODE_ICON_PATH + "OSMisochrone.png",
+)
+@knext.input_table(
+    name="Input Table as Center",
+    description="Input table with geometry.",
+)
+@knext.input_table(
+    name="Input Table as road network",
+    description="Input road network with LineString geometry.",
+)
+@knext.output_table(
+    name="Output Table",
+    description="Output table with isochrone geometry.",
+)
+class IsochroneMap:
+    """
+    This node can calculate the isochrone map for the input point  based on the input road network and its travel cost column.
+    It first snaps input point to the road work, and uses the function
+    [ego_graph](https://networkx.org/documentation/stable/reference/generated/networkx.generators.ego.ego_graph.html)
+    in [NetworkX](https://networkx.org/) to isochrone for the weighted (time or distance) graph.
+
+    The input value for interval list should be cautious, it should be in a reasonable boundary of the road network.
```

Comment:
_The input value for interval list should be cautious_ do you mean something like
_The input value for interval list should be selected carefully_?

Reply:
yes:) I revise that